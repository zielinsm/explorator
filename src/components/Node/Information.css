.Information__Header {
    padding-top: 0 !important;
    margin-bottom: 5px !important;
}
