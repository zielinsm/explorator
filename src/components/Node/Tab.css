.Tab__row {
    padding-bottom: 1em;
}

.Tab__field {
    word-wrap: break-word;
    white-space: pre-wrap;
    max-height: 100px;
    overflow: auto;
}
