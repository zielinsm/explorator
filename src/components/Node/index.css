.Node__no-results {
    padding-top: 1em !important;
    padding-bottom: 1em !important;
}
