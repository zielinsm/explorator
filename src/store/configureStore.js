import { applyMiddleware, createStore } from "redux";
import thunk from "redux-thunk";

import rootReducer from "../reducers";

function configureStore(preloadedState) {
  const store = createStore(
    rootReducer,
    preloadedState,
    applyMiddleware(thunk)
  );
  return store;
}

export default configureStore;
