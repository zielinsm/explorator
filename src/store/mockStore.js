import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const mockSearch = [
  {
    name: "mockNode",
    fingerprint: "MOCKNODE",
    address: "000.000.000.000",
    running: true
  },
  {
    name: "mockRelay",
    fingerprint: "MOCKRELAY",
    address: "000.000.000.000",
    running: true
  }
];

const mockNode = {
  name: "Name",
  id: "FINGERPRINT",
  or_addr: ["000.000.000.000"],
  dir_addr: "111.111.111.111",
  up: true,
  details: ["Mock", "Running"],
  country: "Mock",
  as: {
    number: "AS194",
    name: "Mock Networks"
  },
  weight: 10000,
  host: "mock-relay.net",
  bandwidth: {
    rate: 10000,
    burst: 10000,
    observed: 10000,
    advertised: 10000
  },
  policy: ["reject *:*"],
  policy_summary_v4: { reject: ["1-65535"] },
  policy_summary_v6: { reject: ["1-65535"] },
  family_alleged: ["ALLEGED"],
  family_effective: ["EFFECTIVE"],
  info: "Contact",
  tor: "Tor 0.3.2.10 on Mock"
};

const store = mockStore({
  searchReducer: {
    isLoading: false,
    data: mockSearch
  },
  nodeReducer: {
    isLoading: false,
    data: mockNode
  }
});

export default store;
