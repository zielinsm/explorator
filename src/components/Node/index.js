import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Breadcrumb, Header, Loader, Segment } from "semantic-ui-react";

import { fetchNode } from "../../actions";
import isEmpty from "../../helpers/isEmpty";
import Information from "./Information";
import "./index.css";

function Navigation(props) {
  const { nodeName } = props;
  const resultsLink = `/results/${nodeName}`;

  return (
    <Breadcrumb className="Node__Breadcrumbs">
      <Breadcrumb.Section>
        <Link to="/">Search</Link>
      </Breadcrumb.Section>
      <Breadcrumb.Divider />
      <Breadcrumb.Section>
        Results: <Link to={resultsLink}>{nodeName}</Link>
      </Breadcrumb.Section>
      <Breadcrumb.Divider />
      <Breadcrumb.Section active>Node</Breadcrumb.Section>
    </Breadcrumb>
  );
}

function NoResults() {
  return (
    <div className="Node__no-results">
      <Header as="h5" textAlign="center">
        Such fingerprint doesn't exist. Try again.
      </Header>
    </div>
  );
}

class Node extends Component {
  constructor(props) {
    super(props);
    this.routeParameter = this.props.match.params.id;
  }

  componentDidMount() {
    const { fetchNode } = this.props;
    if (this.routeParameter) {
      fetchNode(this.routeParameter);
    }
  }

  assignSegmentColor(data) {
    if (!isEmpty(data)) {
      return data.up ? "green" : "red";
    } else {
      return "black";
    }
  }

  render() {
    const { nodeReducer } = this.props;
    const { data, isLoading } = nodeReducer;

    if (!isLoading) {
      return (
        <div>
          <Navigation nodeName={data.name} />
          <Segment color={this.assignSegmentColor(data)} className="Node">
            {!isEmpty(data) ? <Information results={data} /> : <NoResults />}
          </Segment>
        </div>
      );
    } else {
      return <Loader active>Fetching data...</Loader>;
    }
  }
}

const mapState = function mapStateToProps(state) {
  const { nodeReducer } = state;
  return { nodeReducer };
};

export default connect(
  mapState,
  { fetchNode }
)(Node);
