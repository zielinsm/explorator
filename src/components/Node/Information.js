import React from "react";
import { Header, Divider, Label, Tab as Switch } from "semantic-ui-react";
import filesize from "filesize";
import "./Information.css";

import Tab from "./Tab";

const Information = props => {
  const { results } = props;

  const {
    name,
    id,
    or_addr,
    dir_addr,
    details,
    country,
    as,
    weight,
    host,
    bandwidth,
    policy,
    family_alleged,
    family_effective,
    info,
    tor
  } = results;

  const basicTab = [
    { label: "OR Addresses", value: or_addr },
    { label: "Directory Address", value: dir_addr },
    { label: "Hostnames", value: host },
    { label: "Country", value: country },
    { label: "AS Name", value: as.name },
    { label: "Consensus weight", value: weight },
    { label: "Platform", value: tor },
    { label: "Contact", value: info }
  ];

  const bandwidthTab = [
    { label: "Bandwidth Rate", value: `${filesize(bandwidth.rate)}/s` },
    { label: "Bandwidth Burst", value: `${filesize(bandwidth.burst)}/s` },
    {
      label: "Observed Bandwidth",
      value: `${filesize(bandwidth.observed)}/s`
    },
    {
      label: "Advertised Bandwidth",
      value: `${filesize(bandwidth.advertised)}/s`
    }
  ];

  const policiesTab = [{ label: "Exit policy", value: policy }];

  const familyTab = [
    { label: "Alleged family", value: family_alleged },
    { label: "Effective family", value: family_effective }
  ];

  const panes = [
    {
      menuItem: "Basic",
      render: () => <Tab data={basicTab} />
    },
    {
      menuItem: "Bandwidth",
      render: () => <Tab data={bandwidthTab} />
    },
    {
      menuItem: "Policies",
      render: () => <Tab data={policiesTab} />
    },
    {
      menuItem: "Family",
      render: () => <Tab data={familyTab} />
    }
  ];

  return (
    <div>
      <Header
        as="h3"
        content={name}
        subheader={id}
        className="Information__Header"
      />
      {details.map((element, index) => (
        <Label size="tiny" key={index}>
          {element}
        </Label>
      ))}
      <Divider />
      <Switch menu={{ secondary: true }} panes={panes} />
    </div>
  );
};

export default Information;
