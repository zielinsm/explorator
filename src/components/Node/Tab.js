import React from "react";
import { Grid, Header } from "semantic-ui-react";

import "./Tab.css";

const convertArray = function convertArrayToAString(element) {
  if (Array.isArray(element)) {
    let formatted = "";
    element.forEach(value => {
      formatted += `${value} \n`;
    });
    return formatted;
  } else {
    return element;
  }
};

const elementCheck = function checkElementForQuirks(value) {
  if (value === undefined || value === null) {
    return "Not defined";
  } else {
    return convertArray(value);
  }
};

function Tab(props) {
  const { data } = props;

  const renderRow = function renderDefaultRow(data, index) {
    const value = elementCheck(data.value);
    const { label } = data;

    return (
      <Grid.Row className="Tab__row" key={index}>
        <Header sub>{label}</Header>
        <span className="Tab__field">{value}</span>
      </Grid.Row>
    );
  };

  return data.map((element, index) => renderRow(element, index));
}

export default Tab;
