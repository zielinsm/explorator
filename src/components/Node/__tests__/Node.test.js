import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import store from "../../../store/mockStore";

import Node from "../index";

describe("Node container:", () => {
  let wrapper;
  const component = (
    <Provider store={store}>
      <MemoryRouter>
        <Node match={{ params: { id: "FINGERPRINT" } }} />
      </MemoryRouter>
    </Provider>
  );

  it("snapshot matches", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
