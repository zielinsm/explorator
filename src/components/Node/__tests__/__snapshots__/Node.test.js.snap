// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node container: snapshot matches 1`] = `
<div>
  <div
    className="ui breadcrumb Node__Breadcrumbs"
  >
    <div
      className="section"
      href={undefined}
      onClick={[Function]}
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
    <div
      className="divider"
    >
      /
    </div>
    <div
      className="section"
      href={undefined}
      onClick={[Function]}
    >
      Results: 
      <a
        href="/results/Name"
        onClick={[Function]}
      >
        Name
      </a>
    </div>
    <div
      className="divider"
    >
      /
    </div>
    <div
      className="active section"
      href={undefined}
      onClick={[Function]}
    >
      Node
    </div>
  </div>
  <div
    className="ui green segment Node"
  >
    <div>
      <h3
        className="ui header Information__Header"
      >
        Name
        <div
          className="sub header"
        >
          FINGERPRINT
        </div>
      </h3>
      <div
        className="ui tiny label"
        onClick={[Function]}
      >
        Mock
      </div>
      <div
        className="ui tiny label"
        onClick={[Function]}
      >
        Running
      </div>
      <div
        className="ui divider"
      />
      <div>
        <div
          className="ui secondary menu"
        >
          <a
            className="active item"
            onClick={[Function]}
          >
            Basic
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Bandwidth
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Policies
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Family
          </a>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            OR Addresses
          </div>
          <span
            className="Tab__field"
          >
            000.000.000.000 
            
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            Directory Address
          </div>
          <span
            className="Tab__field"
          >
            111.111.111.111
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            Hostnames
          </div>
          <span
            className="Tab__field"
          >
            mock-relay.net
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            Country
          </div>
          <span
            className="Tab__field"
          >
            Mock
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            AS Name
          </div>
          <span
            className="Tab__field"
          >
            Mock Networks
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            Consensus weight
          </div>
          <span
            className="Tab__field"
          >
            10000
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            Platform
          </div>
          <span
            className="Tab__field"
          >
            Tor 0.3.2.10 on Mock
          </span>
        </div>
        <div
          className="row Tab__row"
        >
          <div
            className="ui sub header"
          >
            Contact
          </div>
          <span
            className="Tab__field"
          >
            Contact
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
